engine/src/main/java/io/zeebe/engine/processing/common/CatchEventBehavior.java

```diff
@@ -144,16 +152,24 @@ public void subscribeToTimerEvent(
       final long processDefinitionKey,
       final DirectBuffer handlerNodeId,
       final Timer timer,
-      final TypedCommandWriter commandWriter) {
+      final Consumer<SideEffectProducer> sideEffects) {
+    final long dueDate = timer.getDueDate(ActorClock.currentTimeMillis());
     timerRecord.reset();
     timerRecord
         .setRepetitions(timer.getRepetitions())
-        .setDueDate(timer.getDueDate(ActorClock.currentTimeMillis()))
+        .setDueDate(dueDate)
         .setElementInstanceKey(elementInstanceKey)
         .setProcessInstanceKey(processInstanceKey)
         .setTargetElementId(handlerNodeId)
         .setProcessDefinitionKey(processDefinitionKey);
-    commandWriter.appendNewCommand(TimerIntent.CREATE, timerRecord);
+
+    final long timerKey = keyGenerator.nextKey();
+    sideEffects.accept(
+        () -> {
+          dueDateTimerChecker.scheduleTimer(dueDate);
+          return true;
+        });
```

Comment:
We'll move the logic back after migrating all processors which subscribe to timers (see https://github.com/camunda-cloud/zeebe/issues/6589), so this isn't necessary anymore.

However I think we may still need the side effect logic - we only execute side effects after we've updated the state, and we don't want to schedule a timer before we've updated the state, no? I mean, I guess the worst case is you schedule a timer and it has nothing to do, so maybe it's fine.

Reply:
> However I think we may still need the side effect logic - we only execute side effects after we've updated the state, and we don't want to schedule a timer before we've updated the state, no? I mean, I guess the worst case is you schedule a timer and it has nothing to do, so maybe it's fine.

Good point. But it may only happen for very small durations (i.e. `PT0S`). Please add a comment with the reasoning.